test/cloudformation/cloudformation_test.go

```diff
@@ -0,0 +1,177 @@
+package cloudformation
+
+import (
+	"context"
+	"flag"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric/dimension"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation/types"
+	uuid2 "github.com/google/uuid"
+	"log"
+	"os"
+	"testing"
+	"time"
+)
+
+const (
+	timeOut       = 15
+	instanceIdKey = "InstanceId"
+	namespace     = "CFMetricTest"
+)
+
+var (
+	dimensionProviders = []dimension.IProvider{
+		&dimension.CustomDimensionProvider{Provider: dimension.Provider{}},
+	}
+	dimensionFactory = dimension.Factory{Providers: dimensionProviders}
+	packagePath      = flag.String("package_path", "", "path to download package")
+	iamRole          = flag.String("iam_role", "", "iam role for cf ec2 instance")
+	keyName          = flag.String("key_name", "", "key name for ec2 instance")
+	metricName       = flag.String("metric_name", "", "metric to look for")
+)
+
+func TestCloudformation(t *testing.T) {
+	log.Printf("Package path %s iam role %s key name %s metric name %s", *packagePath, *iamRole, *keyName, *metricName)
+	cxt := context.Background()
+	stackName := createStackName()
+
+	client := createCloudwatchClient(cxt)
+
+	startStack(cxt, stackName, client)
+
+	instanceId := findStackInstanceId(cxt, stackName, client)
+
+	for i := 0; i < timeOut; i++ {
+
+	}
+	// Sleep so metrics can be added to cloudwatch
+	log.Printf("Sleep for one minute to collect metrics")
+	time.Sleep(time.Minute)
+
+	dims, failed := dimensionFactory.GetDimensions([]dimension.Instruction{
+		{
+			Key:   instanceIdKey,
+			Value: dimension.ExpectedDimensionValue{Value: aws.String(instanceId)},
+		},
+	})
+
+	if len(failed) > 0 {
+		deleteStack(cxt, stackName, client)
+		t.Fatalf("Failed to generate dimensions")
+		return
+	}
+
+	fetcher := metric.MetricValueFetcher{}
+	values, err := fetcher.Fetch(namespace, *metricName, dims, metric.AVERAGE, test_runner.HighResolutionStatPeriod)
+	if err != nil {
+		deleteStack(cxt, stackName, client)
+		t.Fatalf("Failed to find metric %s namespace %s dimension %v", *metricName, namespace, dims)
+	}
+
+	if !metric.IsAllValuesGreaterThanOrEqualToExpectedValue(*metricName, values, 0) {
+		deleteStack(cxt, stackName, client)
+		t.Fatalf("Metric name %s has a negative value", *metricName)
+	}
+
+	deleteStack(cxt, stackName, client)
+}
+
+func createStackName() string {
+	// Create stack name
+	uuid, err := uuid2.NewUUID()
+	if err != nil {
+		log.Fatalf("Failed to create uuid error %v", err)
+	}
+	stackName := "cfTestStack" + uuid.String()[0:5]
+	return stackName
```

Comment:
nit: could combine line 89 and 90 into:
```
return "cfTestStack" + uuid.String()[0:5]
```

Might want to add a comment why you chose not to use the full UUID?
Was it for readability? Is there a limit to stack name in AWS?
I assume it will be a problem if the stack name currently exists.

Reply:
// Create stack name
	// Only adding the first 5 char for readability
	// Chance of conflict is 1 in 35^5 or about 1 in 52.5 million